### PR TITLE
[entsoe] Refactor HTTP error handling

### PR DESCRIPTION
**Pull request by jlaur**
The primary objective of this pull request is to improve handling of HTTP error code 401 ("Unauthorized") by eliminating exception message string comparison - see https://github.com/openhab/openhab-addons/pull/17416#discussion_r1759837495.

Additionally, we will now provide "openHAB" with version number in the `User-Agent` header. This will make it possible to reach out to us in case of any issues.

And last, but not least, `InterruptedException` will no longer be swallowed by `HttpUtil` and rethrown as `IOException`. This allows us rethrow `InterruptedException` all the way up the call chain. This will ensure that the thread interruption request is respected immediately, rather than masking the exception by catching it prematurely.

### CODE REVIEW

**Comment by jlaur** [comment]
@kaikreuzer - can you add @jmelhus to the contributor team, so he will be added as reviewer for ENTSO-E pull requests? He is our new binding maintainer. 🙂

**Comment by jlaur** [comment]
> Sure thing - done! Welcome @jmelhus 😄 

Hmm, does it usually take time to come into effect, because I still don't see him [here](https://github.com/orgs/openhab/teams/contributors) and still can't request review?

**Comment by kaikreuzer** [comment]
He is invited and needs to accept the invitation before you can see him there.

**Comment by jlaur** [comment]
> He is invited and needs to accept the invitation before you can see him there.

Ah, thanks!

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM. 

Will wait to give @jmelhus some time to confirm.

**Review by jmelhus — APPROVED** [review]
Thanks, nice additions, looks good to me.

**Comment by jlaur** [comment]
> Thanks, nice additions, looks good to me.

Thanks for reviewing. What a coincidence: I marked the PR as draft one minute before your approval. The reason was that I was looking into #17668 and started doing a fix when realizing that the `Request` and `Client` class in **entsoe** was taken from **fmiweather**. 😄 So I was doing the same refactoring there and noticed that I had missed the HTTP error code in the exception message. This is fixed now.

**Pull request merged**
